packages/notification-center/src/store/notifications-provider.context.tsx

```diff
@@ -1,195 +1,129 @@
-import React, { useState } from 'react';
-import { useApi } from '../hooks';
-import { ButtonTypeEnum, IMessage, MessageActionStatusEnum } from '@novu/shared';
-import { NotificationsContext } from './notifications.context';
-import { useFeed } from '../hooks/use-feed.hook';
-
-export function NotificationsProvider({ children }: { children: React.ReactNode }) {
```

Comment:
I did a lot of changes here, but mainly removed the logic that was managing multiple notifications pages assigned to a particular store/feed tab. 
Basically, all of it is cached by the `react-query` so we don't have to do that manually. The only thing we should do is set the current "store", then whenever it changes it will call the `useInfiniteQuery` hook to fetch the notifications.